### ISSUE TITLE
Camera Calibration Problem

### ISSUE THREAD

**Issue by Nick-0814** [tu-darmstadt-ros-pkg/image_projection#11]
Hi Martin, I'm sorry to bother you. 
I recently purchased an insta360 air device, and I would like to ask what open source tool you used to calibrate this device? Is there any good way?

Best regards,
Nick

**Comment by Martin-Oehler** [comment]
Hello Nick,
[Kalibr](https://github.com/ethz-asl/kalibr) has been used for calibration with the omni-radtan model. You can use the calibration provided in my [demo](https://github.com/tu-darmstadt-ros-pkg/image_projection/tree/master/image_projection_demo/config) for testing.

Best regards,
Martin

**Comment by Nick-0814** [comment]
Thanks for your reply, Martin.
I would like to know if you have tried calibrating the external parameters between two fisheye cameras of an insta360 air device? Or default that their optical centers coincide? 
I tried to calibrate the exterior parameter with kalibr, but I was always prompted (Cameras are not connected through mutual observations, please check the dataset. Maybe adjust the approval. sync. tolerance.) 
Sorry to bother you again.
Best regards,
Nick

![problem](https://github.com/tu-darmstadt-ros-pkg/image_projection/assets/111438180/14f1a252-4925-46ed-ab6c-4d7af5d5a63b)

**Comment by Martin-Oehler** [comment]
Yes, we use the cameras with external calibration. It is included in the URDF [here](https://github.com/tu-darmstadt-ros-pkg/image_projection/blob/master/image_projection_demo/urdf/insta360.urdf.xacro#L11).

As the error message tells you, you need sufficient mutual observations to properly estimate the external calibration. Since the overlap between the two Insta360 Air images is very small, we fixed two Insta360 Air cameras together with a 90° offset and calibrated both at the same time.

**Comment by Nick-0814** [comment]
> Yes, we use the cameras with external calibration. It is included in the URDF [here](https://github.com/tu-darmstadt-ros-pkg/image_projection/blob/master/image_projection_demo/urdf/insta360.urdf.xacro#L11).
> 
> As the error message tells you, you need sufficient mutual observations to properly estimate the external calibration. Since the overlap between the two Insta360 Air images is very small, we fixed two Insta360 Air cameras together with a 90° offset and calibrated both at the same time.

Hi,Martin.
Sorry to bother you again.
In the last external parameter calibration, I succeeded according to the idea you provided.
Regarding the focal length parameters of the insta360 air device, I would like to ask if there are any records in your place? I have talked to the technicians of insta company, but they did not offer me any help.
Best regards,
Nick

**Comment by Martin-Oehler** [comment]
kalibr performs intrinsic and extrinsic calibration at the same time, so depending on your [camera model](https://github.com/ethz-asl/kalibr/wiki/supported-models), you can extract the focal length from the intrinsic calibration parameters. In the provided demo configuration, you can find that [here](https://github.com/tu-darmstadt-ros-pkg/image_projection/blob/master/image_projection_demo/config/insta360_3_omni_radtan_calibration/cameras.yaml#L13).

However, it is unclear to my why you would need the focal length parameter explicitly.
